src/org/opensolaris/opengrok/index/IndexDatabase.java

```diff
@@ -969,14 +985,19 @@ private void indexDown(File dir, String parent, IndexDownArgs args)
                             }
                         }
 
-                        // If the file was not modified, skip to the next one.
-                        if (uidIter != null && uidIter.term() != null
-                                && uidIter.term().bytesEquals(buid)) {
-                            BytesRef next = uidIter.next(); // keep matching docs
-                            if (next == null) {
-                                uidIter = null;
-                            }
-                            continue;
+                        /**
+                         * If the file was not modified, probably skip to the
+                         * next one.
+                         */
+                        if (uidIter != null && uidIter.term() != null &&
+                            uidIter.term().bytesEquals(buid)) {
+                            boolean chkres = chkFields(file, path);
+                            if (!chkres) removeFile(false);
```

Comment:
Missing curly braces after `if`, it is a good practice to use them always. Good code:

```
if (!chkres) {
    removeFile(false);
}
```

Reply:
I disagree that it's bad practice for terse, single-line if-statements. This is a style discussion the project will have to have.